src/main/webapp/assets/js/feedScript.js

```diff
@@ -605,16 +614,204 @@ function closeModal() {
  * @return {void}
  */
 async function submitModal() {
-  const collegeId = (new URLSearchParams(window.location.search)).get('collegeid');
+  // Disable multiple submissions.
+  submitModalButton.disabled = true;
 
-  // If one of the fields is empty, don't submit.
-  // Uses formElement.length - 1 to exclude the button element.
+  if (validateModal()) {
+    checkLocationAndSubmit();
+  } else {
+    submitModalButton.disabled = false;
+  }
+}
+
+/**
+ * Goes through the form elements and marks the invalid inputs.
+ * Returns whether all the inputs are valid.
+ * @return {boolean}
+ */
+function validateModal() {
+  const invalidIds = [];
+  const errorMessages = [];
   const formElements = modalForm.elements;
+
+  disableInjection(formElements);
+  validateModalText(invalidIds, errorMessages, formElements);
+  validateModalDate(invalidIds, errorMessages, formElements);
+  validateModalTime(invalidIds, errorMessages, formElements);
+  markInvalidInputs(invalidIds, errorMessages, formElements);
+
+  if (invalidIds.length === 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Goes through the elements and replaces the tags with their HTML equivalent.
```

Comment:
Could you explain why we need to disableInjection as well?

Reply:
Good point, I'll elaborate more in the comment!